src/coreComponents/mesh/MeshLevel.cpp

```diff
@@ -106,401 +107,161 @@ MeshLevel::MeshLevel( string const & name,
 
       newSubRegion.resize( sourceSubRegion.size() );
 
-      arrayView2d< localIndex const > const & elemToFaces = sourceSubRegion.faceList();
-
-      arrayView2d< localIndex const, cells::NODE_MAP_USD > const elemsToNodesSource = sourceSubRegion.nodeList().toViewConst();
-      array2d< localIndex, cells::NODE_MAP_PERMUTATION > & elemsToNodesNew = newSubRegion.nodeList();
-
-      array2d< localIndex > & elemToFacesNew = newSubRegion.faceList();
-      elemToFacesNew = elemToFaces;
-
-      // Copy a new elemCenter map from the old one
+      // copy new elemCenter map from source
       array2d< real64 > & elemCenterNew = newSubRegion.getElementCenter();
       arrayView2d< real64 const > const elemCenterOld = sourceSubRegion.getElementCenter();
       elemCenterNew = elemCenterOld;
-      for( localIndex elem = 0; elem < elemsToNodesNew.size( 0 ); ++elem )
-      {
-        for( localIndex a = 0; a < 3; ++a )
-        {
-          elemCenterNew[elem][a] = elemCenterOld[elem][a];
-        }
-      }
-
-      // Fill a temporary table which knowing the global number of a degree of freedom and a face, gives you the local number of this degree
-      // of freedom on the considering face
-      array2d< localIndex > localElemToLocalFace( 6, numNodesPerElem );
-
-      //Init arrays
-      for( localIndex i = 0; i < 6; ++i )
-      {
-        for( localIndex j = 0; j < numNodesPerElem; ++j )
-        {
-          localElemToLocalFace[i][j]=-1;
-        }
-
-      }
-
-
-
-      for( localIndex i = 0; i < faceToNodeMapSource.size(); ++i )
-      {
-        for( localIndex j = 0; j < numNodesPerFace; ++j )
-        {
-          faceToNodeMapNew[i][j] = -1;
-        }
-
-      }
-
-      //Face 0
-      for( localIndex i = 0; i < order+1; i++ )
-      {
-        for( localIndex j = 0; j <order+1; j++ )
-        {
-          localElemToLocalFace[0][i + numNodesPerFace * j] = i + (order+1)*j;
-        }
 
-      }
-
-      //Face 1
-      for( localIndex i = 0; i < order+1; i++ )
-      {
-        for( localIndex k = 0; k < order+1; k++ )
-        {
-          localElemToLocalFace[1][k+(order+1)*i] = k + (order+1)*i;
-        }
-
-      }
-
-      //Face 2
-      for( localIndex k = 0; k < order+1; k++ )
-      {
-        for( localIndex j = 0; j < order+1; j++ )
-        {
-          localElemToLocalFace[2][k * numNodesPerFace +j*(order+1)] = j + (order+1)*k;
-        }
-
-      }
-
-      //Face 3
-      for( localIndex j = 0; j < order+1; j++ )
-      {
-        for( localIndex k = 0; k < order+1; k++ )
-        {
-          localElemToLocalFace[3][order +k*(order+1)+j * numNodesPerFace] = k + (order+1)*j;
-        }
+      // copy the elements-to-faces map from source
+      arrayView2d< localIndex const > const & elemsToFacesSource = sourceSubRegion.faceList();
+      array2d< localIndex > & elemsToFacesNew = newSubRegion.faceList();
+      elemsToFacesNew = elemsToFacesSource;
```

Comment:
Is this working?
```suggestion
      newSubRegion.faceList() = sourceSubRegion.faceList();
```
If yes, I would adopt it for the for these lines (and the lines around, when it makes sense).

Reply:
Yes, it is working. Thanks Thomas for the suggestion. The changes have been adapted to the code.